### ISSUE TITLE
Variable Substitution misplacing decimal points when mixing region settings

### ISSUE THREAD

**Issue by donnybell** [OctopusDeploy/Issues#6660]
# Prerequisites

- [X] I have verified the problem exists in the latest version
- [X] I have searched [open](https://github.com/OctopusDeploy/Issues/issues) and [closed](https://github.com/OctopusDeploy/Issues/issues?utf8=%E2%9C%93&q=is%3Aissue+is%3Aclosed) issues to make sure it isn't already reported
- [X] I have written a descriptive issue title
- [X] I have linked the original source of this report
- [X] I have tagged the issue appropriately (area/*, kind/bug, tag/regression?)

# The bug

When performing Variable Substitution in a deployment, an Octopus Server with US style numerical region settings (i.e. - 0.1 or 1,000,000.00) deploying to a tentacle with EU style numerical region settings (i.e. - 0,1 or 1.000.000,00) may cause a misplaced decimal point due to the difference in region settings.

# Example

In the project:
Var1 = 0.1

In the target file after variable substitution:
Var1 = 1.0

## What I expected to happen

In the project:
Var1 = 0.1

In the target file after variable substitution:
Var1 = 0.1 (or 0,1)

## Steps to reproduce

"Deploy a Package" with Variable Substitution from Octopus Server (US numerical region settings) to a Tentacle (EU region)

Example Test:
1. Hit "windows" and search "Region Settings"
2. Set country or region to "Netherlands"
3. Open "Additional date, time, or region settings" on the right.
4. Click "Change date, time or number formats"
5. Set first dropdown, format, to "Dutch (netherlands)"
6. Open calamari and run test `JsonFormatVariableReplacerFixture.ShouldReplaceDecimal`

## Affected versions

Octopus Server: ?

## Workarounds

Use the locale specific format for the variable. for example. if you are deploying to a server with an `nl-NL` locale, you should use `0,1`. 

In the project:
Var1 = 0,1

In the target file after variable substitution:
Var1 = 0.1

## Links

Internal:
https://trello.com/c/JDaXDDAu/3905-json-structured-configuration-replacement-doesnt-work-for-decimals-when-os-number-format-is-set-to-dutch-netherlands
https://octopus.zendesk.com/agent/tickets/19860

**Comment by matt-richardson** [comment]
We are closing this as by-design. While there is a mismatch between what gets entered in the UI, and what results in the json file, it is still logical. Our reasoning is that in the `nl-NL` locale, `0,1` is how you write "1 tenth of 1". In the [JSON schema](https://www.json.org/json-en.html), this is formatted as `0.1`.

While we can understand the desire for a change of behavior here, changing this would probably break more scenarios than it fixes.

TLDR: please enter floats in the format required by the target locale.
